### DISCUSSION TITLE
[Feature]: Animation moving along with a curve.

### DISCUSSION THREAD

**Discussion by MrXBear** [fabricjs/fabric.js#9071] [Q&A] [answered]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have searched and referenced existing issues, feature requests and discussions
- [X] I am filing a **FEATURE** request.

### Description

We can move an object by setting a new coordinate and `renderAll`.
Then the object move along with a straight line.

Can we move the object along with a specific curve?
Such as 
```
object.animate({
  path: somePath,
  duration: 1000,
})
```

![path](https://github.com/fabricjs/fabric.js/assets/47053993/61b89893-4a9d-4a86-93f7-f4f8bb4c5f1a)

### Current State

Obejcts moving along with a straight line in animation.

**Comment by ShaMan123** [comment]
Look at path utils

**Comment by ShaMan123** [comment]
I really like the API 

> ```
> object.animate({
>   path: somePath,
>   duration: 1000,
> })
>```

**Comment by ShaMan123** [comment]
@asturur what do you think about this

**Comment by ShaMan123** [comment, marked as answer]
#9077
